packages/react-ui/src/app/builder/builder-page.tsx

```diff
@@ -30,55 +53,77 @@ const BuilderPage = () => {
       state.ExitRun,
       state.selectedStep,
     ]);
-
+  const [isDraggingHandle, setIsDraggingHandle] = useState(false);
+  const rightHandleRef = useAnimateSidebar(
+    useCallback(() => rightSidebar === RightSideBarType.NONE, [rightSidebar]),
+    rightSidebar,
+  );
+  const leftHandleRef = useAnimateSidebar(
+    useCallback(() => leftSidebar === LeftSideBarType.NONE, [leftSidebar]),
+    leftSidebar,
+  );
```

Comment:
```suggestion
  const [isDraggingHandle, setIsDraggingHandle] = useState(false);
  const rightHandleRef = useAnimateSidebar(
    rightSidebar,
  );
  const leftHandleRef = useAnimateSidebar(
    leftSidebar,
  );
```

I don't think useCallback is needed, you already know the values of isSidebarClosed() and in case it's changed, the component will re-render.